core/arch/arm/plat-marvell/armada7080/hal_sec_perf.c

```diff
@@ -0,0 +1,304 @@
+#define RA_SIZE	(TZDRAM_SIZE)
+#define RA_PERM	(ABORT_PERM)
+
+#define TZ_GET_VALID(data, ret)		((ret) = ((data) & (0x1)))
+#define TZ_SET_VALID(data, val)		\
+	do { \
+		(data) &= (~(0x1));     \
+		(data) |= ((val) & 0x1);	\
+	} while (0)
+
+#define TZ_GET_PERM(data, ret)		((ret) = (((data) & (0x3 << 1)) >> 1))
+#define TZ_SET_PERM(data, val)		\
+	do {	\
+		(data) &= (~(0x3 << 1)); \
+		(data) |= (((val) & 0x3) << 1);	\
+	} while (0)
+
+#define TZ_GET_RZ_EN(data, ret)		((ret) = (((data) & (0x1 << 3)) >> 3))
+#define TZ_SET_RZ_EN(data, val)		\
+	do {	\
+		(data) &= (~(0x1 << 3)); \
+		(data) |= (((val) & 0x1) << 3);	\
+	} while (0)
+
+#define TZ_GET_AREA_LEN_CODE(data, ret)	((ret) = (((data) & (0x1F << 7)) >> 7))
+
+#define TZ_SET_AREA_LEN_CODE(data, val)	\
+	do {	\
+		(data) &= (~(0x1F << 7));	\
+		(data) |= (((val) & 0x1F) << 7);		\
+	} while (0)
+
+#define TZ_GET_START_ADDR_L(data, ret)	\
+	((ret) = (((data) & 0xFFFFF000)))
+
+#define TZ_SET_START_ADDR_L(data, val)		\
+	do {	\
+		(data) &= (~0xFFFFF000);		\
+		(data) |= (((val) & 0xFFFFF000));	\
+	} while (0)
+
+#define TZ_GET_UR_PERM(data, val)	((ret) = (((data) & (0x3 << 4)) >> 4))
+#define TZ_SET_UR_PERM(data, val)	\
+	do {	\
+		(data) &= (~(0x3 << 4)); \
+		(data) |= (((val) & 0x3) << 4);	\
+	} while (0)
+
+#define TZ_GET_UR_RZ_EN(data, val)		\
+	((ret) = (((data) & (0x1 << 6)) >> 6))
+
+#define TZ_SET_UR_RZ_EN(data, val)		\
+	do {	\
+		(data) &= (~(0x1 << 6)); \
+		(data) |= (((val) & 0x1) << 6);	\
+	} while (0)
+
+ /* armada mini region size is 1M */
+#define RANGE_SIZE_TO_CODE(size, code, i)	\
+	do {	\
+		(code) = INVALID_SIZE_CODE;	\
+		for ((i) = 8; (i) <= 0x1f; (i)++) {	   \
+		if (((uint32_t)0x1 << (i)) == ((size) >> 12)) { \
+			(code) = (i);	\
+			break;	\
+			}	\
+		}	\
+	} while (0)
+
+#define RANGE_CODE_TO_SIZE_K(code, sizek)	\
+	((sizek) = ((4) << (code)))	\
+
+#define TZ_LOCK_MC(x)		\
+	do {	\
+		(x) = read32(MCU_MC_CONTROL_0_REG);	\
+		(x) |= (TRUSTZONE_LOCK);	\
+		 write32((x), MCU_MC_CONTROL_0_REG);	\
+	} while (0)
+
+#define _IS_ALIGNED(_addr, _algn)	(!((_addr) & ((_algn) - 1)))
+
+register_phys_mem(MEM_AREA_IO_SEC, MCU_BASE, CORE_MMU_DEVICE_SIZE);
+register_phys_mem(MEM_AREA_IO_SEC, MC_SCR_REGISTER, CORE_MMU_DEVICE_SIZE);
+
+static int32_t _find_valid_range(void)
+{
+	uint32_t i;
+	uint32_t tmp;
+	uint32_t valid;
+
+	for (i = 0; i < MAX_RANGE_NUM; i++) {
+		tmp = read32(MCU_TZ_RANGE_LOW_REG(i));
+		TZ_GET_VALID(tmp, valid);
+		if (!(valid))
+			return i;
+	}
+	return -1;
+}
+
+static int32_t set_range(uint32_t addr, uint32_t size, uint32_t perm)
+{
+	uint32_t data;
+	uint32_t sizecode;
+	int32_t valid_range;
+	uint32_t i;
+
+	if (!_IS_ALIGNED(addr, SIZE_1M)) {
+		EMSG("ERR: region addr(0x%x) is not aligned with 1M!", addr);
+		return -1;
+	}
+
+	if (!_IS_ALIGNED(size, SIZE_1M)) {
+		EMSG("ERR: region size(0x%x) is not aligned with 1M!", size);
+		return -1;
+	}
+
+	if (!_IS_ALIGNED(addr, size)) {
+		EMSG("ERR: region size(0x%x) not align with addr(0x%x)"
```

Comment:
minor: 
- prefer the `,` at the end of the previous line.
- use `"... 0x%" PRIx32` rather than `" ... 0x%x"`, here and below.